src/main/java/org/spongepowered/api/util/Property.java

```diff
@@ -0,0 +1,226 @@
+            }
+        },
+
+        /**
+         * Use the <code>.equals()</code> method but match if the result is
+         * false
+         */
+        NOTEQUAL {
+            @Override
+            protected boolean apply(Property<?, ?, ?> operand1, Property<?, ?, ?> operand2) {
+                return !operand2.equals(operand1);
+            }
+        },
+
+        /**
+         * Use <code>.compareTo()</code> amd match if target is greater than
+         * supplied operand
+         */
+        GREATER {
+            @Override
+            protected boolean apply(Property<?, ?, ?> operand1, Property<?, ?, ?> operand2) {
+                @SuppressWarnings({"unchecked", "rawtypes"})
+                int result = ((Comparable) operand2).compareTo(operand1);
+                return result > 0;
+            }
+        },
+
+        /**
+         * Use <code>.compareTo()</code> amd match if target is greater than
+         * or equal to supplied operand
+         */
+        GEQUAL {
+            @Override
+            protected boolean apply(Property<?, ?, ?> operand1, Property<?, ?, ?> operand2) {
+                @SuppressWarnings({"unchecked", "rawtypes"})
+                int result = ((Comparable) operand2).compareTo(operand1);
+                return result >= 0;
+            }
+        },
+
+        /**
+         * Use <code>.compareTo()</code> amd match if target is less than the
+         * supplied operand
+         */
+        LESS {
+            @Override
+            protected boolean apply(Property<?, ?, ?> operand1, Property<?, ?, ?> operand2) {
+                @SuppressWarnings({"unchecked", "rawtypes"})
+                int result = ((Comparable) operand2).compareTo(operand1);
+                return result < 0;
+            }
+        },
+
+        /**
+         * Use <code>.compareTo()</code> amd match if target is less than or
+         * equal to the supplied operand
+         */
+        LEQUAL {
+            @Override
+            protected boolean apply(Property<?, ?, ?> operand1, Property<?, ?, ?> operand2) {
+                @SuppressWarnings({"unchecked", "rawtypes"})
+                int result = ((Comparable) operand2).compareTo(operand1);
+                return result <= 0;
+            }
+        };
+
+        /**
+         * <p>Compare the two operands by applying this operator <em>infix</em>
+         * with respect to them. For example, if this object is
+         * {@link Operator#GREATER} then calling this method with
+         * <code>object1</code> and <code>object2</code> would effectively
+         * return:</p>
+         *
+         * <blockquote> <code>returnValue = object1 &gt; object2;</code>
+         * </blockquote>
+         *
+         * @param operand1 First operand
+         * @param operand2 Second operand
+         * @return First operand's comparison to second
+         */
+        public final boolean compare(@Nullable Property<?, ?, ?> operand1, @Nullable Property<?, ?, ?> operand2) {
```

Comment:
Changing these generics changes the properties in Inventory API. Considering that the Inventory Properties are supposed to be comparable to other properties.